api/grpcserver.go

```diff
@@ -225,14 +232,19 @@ func (svr *GRPCServer) GetReceiptByAction(ctx context.Context, in *iotexapi.GetR
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	receipt, blkHash, err := svr.coreService.ReceiptByAction(actHash)
+	receipt, err := svr.coreService.ReceiptByActionHash(actHash)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.NotFound, err.Error())
 	}
+	blkHash, err := svr.coreService.GetBlockHashByActionHash(actHash)
```

Comment:
this is not efficient. you may consider to use `receipt.BlockHeight` and fetch the block hash by block height.